### DISCUSSION TITLE
EnkPi range of e-ink displays

### DISCUSSION THREAD

**Discussion by midnightcoder2** [peterhinch/micropython-nano-gui#58] [Ideas]
Hi, I'm trying to write a driver for the EnkPi e-ink display.  It's a display with a pico and other stuff built into it.  The py file already contains much of the code similar to what is in the driver files, but I'm struggling to re-write this for the enkpi.  Any help appreciated.

https://github.com/sbcshop/EnkPi_7.5_Software/tree/main

**Comment by peterhinch** [comment]
[The existing driver](https://github.com/sbcshop/EnkPi_7.5_Software/blob/main/lib/EnkPi_7in5.py) is incompatible with `nano-gui` because the `E_paper` class is not derived from `FrameBuffer`. It also requires 96KiB of buffer space which is very large. I have avoided writing drivers for mult-color Epaper displays because the refresh time is usually long, and large ones require big buffers.

I therefore don't see this display as a candidate for nano-gui. My advice is either to use this display as suggested by its manufacturer or to use `nano-gui` with one of the many supported displays.
